inference-engine/src/mkldnn_plugin/nodes/common/cpu_convert.cpp

```diff
@@ -46,73 +216,270 @@ struct ConvertContext {
 };
 
 template<typename T>
-struct ConvertPrecision {
-    using src_t = typename std::tuple_element<0, T>::type;
-    using dst_t = typename std::tuple_element<1, T>::type;
+struct ConvertPrecision;
+
+template<typename src_t, typename dst_t>
+struct ConvertPrecision<std::tuple<src_t, dst_t>> {
+    void operator()(ConvertContext & ctx) {
+        auto src = static_cast<const src_t *>(ctx.srcPtr);
+        auto dst = static_cast<dst_t *>(ctx.dstPtr);
+        parallel_for(ctx.size, [&](size_t i) {
+            dst[i] = static_cast<dst_t>(src[i]);
+        });
+        ctx.converted = true;
+    }
+};
 
+template<typename src_t>
+struct ConvertPrecision<std::tuple<src_t, ov::float16>> {
     void operator()(ConvertContext & ctx) {
-        convert<src_t, dst_t>(ctx.srcPtr, ctx.dstPtr, ctx.size);
+        auto src = static_cast<const src_t *>(ctx.srcPtr);
+        auto dst = static_cast<ov::float16 *>(ctx.dstPtr);
+
+        constexpr size_t batch = 64;
+        const size_t iterations = (ctx.size + batch - 1) / batch;
```

Comment:
minor: div_up from general_utils.h can be used.

Reply:
done